modyn/supervisor/internal/triggers/drift/decision_policy.py

```diff
@@ -41,27 +41,58 @@ def evaluate_decision(self, distance: float) -> bool:
 
 class DynamicDecisionPolicy(DriftDecisionPolicy):
     """Decision policy that will make the binary is_drift decisions based on a
-    dynamic threshold.
+    dynamic threshold."""
+
+    def __init__(self, config: DynamicThresholdCriterion):
+        self.config = config
+        self.score_observations: deque = deque(maxlen=self.config.window_size)
+
+
+class DynamicPercentileThresholdPolicy(DynamicDecisionPolicy):
+    """Dynamic threshold based on a extremeness percentile of the previous
+    distance values.
 
     We compare a new distance value with the series of previous distance values
     and decide if it's more extreme than a certain percentile of the series. Therefore we count the
     `num_more_extreme` values that are greater than the new distance and compare it with the
     `percentile` threshold.
-
-    TODO: we might want to also support some rolling average policy that will trigger if a distance is deviates
-    from the average by a certain amount.
     """
 
-    def __init__(self, config: DynamicThresholdCriterion):
-        self.config = config
-        self.score_observations: deque = deque(maxlen=self.config.window_size)
+    def evaluate_decision(self, distance: float) -> bool:
+        if len(self.score_observations) == 0:
+            self.score_observations.append(distance)
+            return True
+
+        sorted_observations = list(sorted(self.score_observations))
+
+        threshold = sorted_observations[
+            min(
+                max(
+                    0,
+                    int(len(sorted_observations) * (1.0 - self.config.percentile)) - 1,
```

Comment:
why len() * perc - 1 instead of just rounding?

Reply:
we compute an index here, they `-1` should just account for the fact the python indexes are 0-based.

I had the following semantics in mind:
- Assume we have 10 observations ready
- user requests 0% significance: `len(sorted_observations) * (1.0 - self.config.percentile) = 10`
- user requests 10% significance: --> `9`
- with the `-1` we get valid indexes

What happens to the cases in between (e.g. `5%`) is a design decision. Rounding is actually desired here! The `-1` is still needed, though.